internal/services/nginx/nginx_configuration_resource_test.go

```diff
@@ -111,6 +111,10 @@ resource "azurerm_nginx_configuration" "test" {
     content      = local.protected_content
     virtual_path = "/opt/.htpasswd"
   }
+
+  lifecycle {
+    ignore_changes = [protected_file]
+  }
```

Comment:
this is indicative of a bug. is the entire block or just the contents not returned?

either way the correct fix is to pull the value over from config/state on read via d.get so this is not required

Reply:
Just an FYI: We have updated the protect files swagger to return some contents (I think just the hash but have to double check) and might be why the block is returning something which you are trying to ignore vi `ignore_changes`?

Instead of doing this, can I please request that you disable the test or the specific portion of the protected files test to avoid losing full coverage and we revisit it?

Reply:
Hi @wuxu92 Can you elaborate what the error is that you shared in this message?

Also what is the issue that you are seeing that you have to use `ignore_changes`?

Reply:
I think I see the problem. The service provider API has moved forward. Earlier on, protected files, once submitted, were not returned but now they are returned without its contents.

That's probably why you see a diff where TF is trying to remove the block as it is not expecting to see it.

Reply:
Given this is an additive change, I think we can safely handle it in terraform.